### DISCUSSION TITLE
How to fix 'unable to verify the first certificate' issue with GitHub Copilot

### DISCUSSION THREAD

**Discussion by ace1959899** [community/community#58299] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

GitHub Copilot could not connect to server. Extension activation failed: "unable to verify the first certificate"

**Comment by the-officialjosh** [comment]
The error message "unable to verify the first certificate" when trying to activate the GitHub Copilot extension in VS Code may be caused by a corporate proxy setup that intercepts secure connections, terminates the secure connection to GitHub, and instead uses a self-signed certificate to act like a TLS middleman. Here are some possible solutions:

🔧 Contact your IT department to see if they can configure the proxy to not intercept secure connections.

🔧 Properly install the proxy certificates on your machine in the OS trust chain. Note that this requires a GitHub Copilot for Business subscription.

🔧 Configure a different proxy that does not intercept secure connections.

If the above solutions do not work, it is possible that something has either corrupted the set of SSL roots on your machine or a third party is intercepting your SSL connections. In this case, you can try the following:

🔧 Uninstall and reload GitHub Copilot.

🔧 Change the version of GitHub Copilot.

🔧 Quit VS Code, turn off the accelerator, and then reopen VS Code.

If none of the above solutions work, it may be necessary to try using GitHub Copilot on a different machine.

**Comment by alrik11es** [comment]
![imagen](https://github.com/community/community/assets/249844/d87549bd-af51-4946-aeaf-24c6100d51c0)

I'm having the same problem on windows. I've tried everything. In mac works well.

**Comment by noweihed** [comment]
I have the same problem on the one computer, i tried everything uninstalled/reinstalled copilot, i even uninstalled Rider twice and still the same problem.   I just tried form my laptop and there is NO Problem on that machine...  it is using v1.2.9... i want to try installing that version on the desktop but i cant figure out how to switch it, it automatically installs 1.5.2

**Comment by lada-olc** [comment, marked as answer]
I added an Ignore rule in ESET SSL/TLS configuration as advised in https://github.com/orgs/community/discussions/120205#discussioncomment-9165225 and Copilot works now.

**Reply by noweihed** [reply]
FINALLY!!! I have been after this for like 3 days ! i guess i have been searching for the wrong terms or in the wrong place!  This worked for me!

**Reply by okirmis** [reply]
Can confirm that this is the solution for us, too.

**Reply by alrik11es** [reply]
Damn ESET🤦🏻‍♂️ 
Can confirm this works for me too. THANKS.
